src/main.rs

```diff
@@ -130,9 +129,9 @@ async fn find_workir(path: String) -> anyhow::Result<PathBuf> {
 /// none, hah :)
 #[instrument]
 async fn extract_device(path: &std::path::Path) -> Option<String> {
-    guard!(let Some(p) = path.as_os_str().to_str() else {return None});
-    guard!(let Some(res) = p.split_terminator('/').last() else {return None});
-    Some(res.to_owned())
+  guard!(let Some(p) = path.as_os_str().to_str() else {return None});
```

Comment:
The question mark operator works with option return types, no need for this guard stuff.

Reply:
? works only on results, not options, unless i dont know something

Reply:
Try it. std::ops::Try is implemented for Option

Reply:
fixed